docs/source/basics/message_transforms.rst

```diff
@@ -0,0 +1,135 @@
+.. _message_transform_usage_label:
+
+==================
+Message Transforms
+==================
+
+Message transforms perform the conversion of raw sample dictionaries from your dataset into torchtune's
+:class:`~torchtune.data.Message` structure. This is where you, as the architect of your data and training pipeline,
+get to place all your data preprocessing and formatting logic without having to worry about the rest of
+the pipeline to tokenization and model inputs. We provide utilities to make this as easy as possible,
+and if your dataset folows a standard format, you can use our built-in utilities to quickly get
+fine-tuning without having to write any data transforms.
+
+
+Configuring message transforms
+------------------------------
+Most of our built-in message transforms contain parameters for controlling input masking (``train_on_input``),
+adding a system prompt (``new_system_prompt``), and changing the expected column names (``column_map``).
+These are exposed in our dataset builders :func:`~torchtune.datasets.instruct_dataset` and :func:`~torchtune.datasets.chat_dataset`
+so you don't have to worry about the message transform itself and can configure this directly from the config.
+You can view the API ref for these builders for more details.
+
+For example, :func:`~torchtune.datasets.instruct_dataset` uses :class:`~torchtune.data.InputOutputToMessages` as the message transform.
```

Comment:
I understand that you are giving context, but its 40 lines of text that dont tell the user how to write a transform. It is showing how to use instruct dataset, which is covered in the other tutorial.

Reply:
will point to instruct docs for the example